KeY crashes after initial launch terminates unconventionally

## Description

> When KeY is launched without an existing .key directory, it creates this directory. Upon termination of the program KeY would write its settings in that directory. If KeY is not terminated through its provided methods, the settings are not written to a file.
When next launching KeY it finds the directory, but cannot load the settings. Thus it fails to start and crashes.

## Reproducible
  
> Is the issue reproducible?
always

### Steps to reproduce 

> Describe the steps needed to reproduce the issue.

1. Ensure there is no .key directory
2. Launch KeY
3. Kill the process
4. Launch KeY again

> What is your expected behavior and what was the actual behavior?

expected: KeY launches and loads standard settings
actual: KeY fails to launch

### Additional information

Caused by: java.lang.NullPointerException: Cannot invoke "de.uka.ilkd.key.settings.Settings.addPropertyChangeListener(java.beans.PropertyChangeListener)" because "settings" is null


---

* Commit: 5b08998f3cc0c0236fbe30abaf51901871d76dbe

--- a/key.core/src/main/java/de/uka/ilkd/key/nparser/ParsingFacade.java
+++ b/key.core/src/main/java/de/uka/ilkd/key/nparser/ParsingFacade.java
@@ -208,7 +208,7 @@ public final class ParsingFacade {
 
     // region configuration
     /**
-     * Parses a the configuration determined by the given {@code file}.
+     * Parses the configuration determined by the given {@code file}.
      * A configuration corresponds to the grammar rule {@code cfile} in the {@code KeYParser.g4}.
      *
      * @param file non-null {@link Path} object
@@ -221,19 +221,20 @@ public final class ParsingFacade {
     }
 
     /**
+     * @param file non-null file to read as configuration
      * @see #parseConfigurationFile(Path)
+     * @throws IOException if the file is not found or not readable.
      */
     public static KeyAst.ConfigurationFile parseConfigurationFile(File file) throws IOException {
         return parseConfigurationFile(file.toPath());
     }
 
     /**
-     * Parses a the configuration determined by the given {@code stream}.
+     * Parses the configuration determined by the given {@code stream}.
      * A configuration corresponds to the grammar rule {@code cfile} in the {@code KeYParser.g4}.
      *
-     * @param file non-null {@link CharStream} object
+     * @param stream non-null {@link CharStream} object
      * @return monad that encapsluate the ParserRuleContext
-     * @throws IOException if the file is not found or not readable.
      * @throws BuildingException if the file is syntactical broken.
      */
     public static KeyAst.ConfigurationFile parseConfigurationFile(CharStream stream) {
@@ -244,20 +245,20 @@ public final class ParsingFacade {
     }
 
     /**
-     * Parses a the configuration determined by the given {@code stream}.
+     * Parses the configuration determined by the given {@code stream}.
      * A configuration corresponds to the grammar rule {@code cfile} in the {@code KeYParser.g4}.
      *
-     * @param file non-null {@link CharStream} object
-     * @return a configration object with the data deserialize from the given file
-     * @throws IOException if the file is not found or not readable.
+     * @param input non-null {@link CharStream} object
+     * @return a configuration object with the data deserialize from the given file
      * @throws BuildingException if the file is syntactical broken.
      */
-    public static Configuration readConfigurationFile(CharStream input) throws IOException {
+    public static Configuration readConfigurationFile(CharStream input) {
         return parseConfigurationFile(input).asConfiguration();
     }
 
     /**
      * @see #readConfigurationFile(CharStream)
+     * @throws IOException if the file is not found or not readable.
      */
     public static Configuration readConfigurationFile(Path file) throws IOException {
         return readConfigurationFile(CharStreams.fromPath(file));
@@ -265,6 +266,7 @@ public final class ParsingFacade {
 
     /**
      * @see #readConfigurationFile(CharStream)
+     * @throws IOException if the file is not found or not readable.
      */
     public static Configuration readConfigurationFile(File file) throws IOException {
         return readConfigurationFile(file.toPath());

--- a/key.ui/src/main/java/de/uka/ilkd/key/gui/colors/ColorSettings.java
+++ b/key.ui/src/main/java/de/uka/ilkd/key/gui/colors/ColorSettings.java
@@ -8,14 +8,11 @@ import java.io.File;
 import java.io.FileWriter;
 import java.io.IOException;
 import java.io.Writer;
-import java.nio.charset.StandardCharsets;
-import java.util.Map;
 import java.util.Objects;
 import java.util.Optional;
 import java.util.Properties;
 import java.util.stream.Stream;
 
-import de.uka.ilkd.key.gui.settings.SettingsManager;
 import de.uka.ilkd.key.settings.AbstractPropertiesSettings;
 import de.uka.ilkd.key.settings.Configuration;
 import de.uka.ilkd.key.settings.PathConfig;
@@ -33,20 +30,10 @@ import org.slf4j.LoggerFactory;
  * @version 1 (10.05.19)
  */
 public class ColorSettings extends AbstractPropertiesSettings {
-    public static final String SETTINGS_FILENAME = "colors.properties";
-    public static final File SETTINGS_FILE =
-        new File(PathConfig.getKeyConfigDir(), SETTINGS_FILENAME);
-
     public static final File SETTINGS_FILE_NEW =
         new File(PathConfig.getKeyConfigDir(), "colors.json");
     private static final Logger LOGGER = LoggerFactory.getLogger(ColorSettings.class);
     private static ColorSettings INSTANCE;
-
-    private ColorSettings(Properties settings) {
-        super("");
-        readSettings(settings);
-        Runtime.getRuntime().addShutdownHook(new Thread(this::save));
-    }
 
     public ColorSettings(Configuration load) {
         super("");
@@ -56,15 +43,17 @@ public class ColorSettings extends AbstractPropertiesSettings {
 
     public static ColorSettings getInstance() {
         if (INSTANCE == null) {
-            if (SETTINGS_FILE.exists()) {
+            if (SETTINGS_FILE_NEW.exists()) {
                 try {
-                    LOGGER.info("Use new configuration format at {}", SETTINGS_FILE_NEW);
-                    return INSTANCE = new ColorSettings(Configuration.load(SETTINGS_FILE_NEW));
+                    LOGGER.info("Load color settings from file {}", SETTINGS_FILE_NEW);
+                    INSTANCE = new ColorSettings(Configuration.load(SETTINGS_FILE_NEW));
+                    return INSTANCE;
                 } catch (IOException e) {
                     LOGGER.error("Could not read {}", SETTINGS_FILE_NEW, e);
                 }
             }
-            return INSTANCE = new ColorSettings(SettingsManager.loadProperties(SETTINGS_FILE));
+            INSTANCE = new ColorSettings(new Configuration());
+            return INSTANCE;
         }
         return INSTANCE;
     }
@@ -94,22 +83,10 @@ public class ColorSettings extends AbstractPropertiesSettings {
     /**
      * Writes the current settings to default location.
      *
-     * @see #SETTINGS_FILE
+     * @see #SETTINGS_FILE_NEW
      */
     public void save() {
-        LOGGER.info("Save color settings to: " + SETTINGS_FILE.getAbsolutePath());
-        try (Writer writer = new FileWriter(SETTINGS_FILE, StandardCharsets.UTF_8)) {
-            Properties props = new Properties();
-            for (Map.Entry<String, Object> entry : properties.entrySet()) {
-                props.setProperty(entry.getKey(), entry.getValue().toString());
-            }
-            props.store(writer, "KeY's Colors");
-            writer.flush();
-        } catch (IOException ex) {
-            ex.printStackTrace();
-        }
-
-        LOGGER.info("Save color settings to: " + SETTINGS_FILE_NEW.getAbsolutePath());
+        LOGGER.info("Save color settings to: {}", SETTINGS_FILE_NEW.getAbsolutePath());
         try (Writer writer = new FileWriter(SETTINGS_FILE_NEW)) {
             var config = new Configuration(properties);
             config.save(writer, "KeY's Colors");

--- a/key.ui/src/main/java/de/uka/ilkd/key/gui/colors/ColorSettingsProvider.java
+++ b/key.ui/src/main/java/de/uka/ilkd/key/gui/colors/ColorSettingsProvider.java
@@ -32,7 +32,7 @@ public class ColorSettingsProvider extends SimpleSettingsPanel implements Settin
         super();
         setHeaderText(getDescription());
         setSubHeaderText(
-            "Color settings are stored in: " + ColorSettings.SETTINGS_FILE.getAbsolutePath());
+            "Color settings are stored in: " + ColorSettings.SETTINGS_FILE_NEW.getAbsolutePath());
         add(new JScrollPane(tblColors));
     }
 

